src/test/java/io/openliberty/tools/intellij/it/SingleModMPProjectTestCommon.java

```diff
@@ -923,6 +924,78 @@ public static void prepareEnv(String projectPath, String projectName) {
                 "prepareEnv. Exit. ProjectName: " + projectName);
     }
 
+    /**
+     * Tests that when a Liberty run configuration is removed, any custom start parameters
+     * linked to the Liberty module are also cleared.
+     */
+    @Test
+    @Video
+    public void testCustomStartParametersClearedOnConfigRemoval() {
+        String testName = "testCustomStartParametersClearedOnConfigRemoval";
+        String absoluteWLPPath = Paths.get(getProjectsDirPath(), getSmMPProjectName(), getWLPInstallPath()).toString();
+
+        // Remove all existing configurations for a clean state.
+        UIBotTestUtils.deleteLibertyRunConfigurations(remoteRobot);
+
+        // Trigger the start with parameters configuration dialog.
+        UIBotTestUtils.runLibertyActionFromLTWDropDownMenu(remoteRobot, "Start...", false, 3);
+
+        // Run the configuration dialog.
+        UIBotTestUtils.runStartParamsConfigDialog(remoteRobot, getStartParamsDebugPort());
+
+        try {
+            // Validate that the project started.
+            TestUtils.validateProjectStarted(testName, getSmMpProjResURI(), getSmMpProjPort(), getSmMPProjOutput(), absoluteWLPPath, false);
+
+            // Expand the Project tree and open server.env file
+            UIBotTestUtils.openFile(remoteRobot, getSmMPProjectName(), "server.env", getServerEnvPath());
+            // Check if the specified Custom debug port is used.
+            TestUtils.checkCustomDebugPort(absoluteWLPPath);
+
+        } catch (IOException e) {
+            System.err.println("Error reading the file: " + e.getMessage());
+
+        } finally {
+            cleanupAndStopServerIfNeeded(absoluteWLPPath);
+        }
+
+        // Cleanup configurations.
+        UIBotTestUtils.deleteLibertyRunConfigurations(remoteRobot);
+
+        // Start dev mode.
+        UIBotTestUtils.runLibertyActionFromLTWDropDownMenu(remoteRobot, "Start", true, 3);
+
+        try {
+            // Validate that the project started.
+            TestUtils.validateProjectStarted(testName, getSmMpProjResURI(), getSmMpProjPort(), getSmMPProjOutput(), absoluteWLPPath, false);
+
+            // Expand the Project tree and open server.env file
+            UIBotTestUtils.openFile(remoteRobot, getSmMPProjectName(), "server.env", getServerEnvPath());
+            // Check if Default debug port is used.
+            TestUtils.checkDefaultDebugPort(absoluteWLPPath);
+
+        } catch (IOException e) {
+            System.err.println("Error reading the file: " + e.getMessage());
+
+        } finally {
+            cleanupAndStopServerIfNeeded(absoluteWLPPath);
+        }
+    }
+
+    private void cleanupAndStopServerIfNeeded(String absoluteWLPPath) {
```

Comment:
Can we move this method to utils or some common classes, so that we can reuse this method in other tests.

Reply:
Moved cleanupAndStopServerIfNeeded into UIBotTestUtils.